geomesa-hbase/geomesa-hbase-rpc/src/main/scala/org/locationtech/geomesa/hbase/rpc/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -134,31 +147,48 @@ object GeoMesaCoprocessor extends LazyLogging {
 
     private val coprocessor = CachedThreadPool.submit(new Runnable() {
       override def run(): Unit = {
+        val callback = new GeoMesaHBaseCallBack(resultQueue)
         try {
           if (!closed.get) {
+            logger.trace(s"Calling htable.coproService (first time): " +
+              s"${ByteArrays.printable(scan.getStartRow)} to ${ByteArrays.printable(scan.getStopRow)}")
             htable.coprocessorService(service, scan.getStartRow, scan.getStopRow, callable, callback)
           }
+          logger.trace(s"Closed: ${closed.get}. Callback.lastRow: ${ByteArrays.printable(callback.lastRow)}")
+
+          // If the scan hasn't been killed and we are not done, then re-issue the request.
+          while (!closed.get() && callback.lastRow != null) {
```

Comment:
[minor] we could avoid some code duplication by setting the callback.lastRow to the scan.startRow up front, then just using the while loop and not making that first call just above this. it would involve an extra null check...

Reply:
I tried to do something like that here: https://github.com/ccri/geomesa/commit/374ca789e43c5daa5b3d62cf9a80c335b588fa78 but the unit tests got into an infinite loop. Is my error obvious?

Reply:
hmm, i don't see anything obvious ¯\_(ツ)_/¯